### ISSUE TITLE
AlphaHPP without 3D Secure (`3DSecureBypass` = `Y`): «Provider unable to process request» / «"ProviderName": "westpac"»

### ISSUE THREAD

**Issue by dmitrii-fediuk** [mage2pro/alphacommercehub#8]
![01](https://user-images.githubusercontent.com/83204/32404063-cec0db72-c159-11e7-9850-491189a89722.png)

## The server's response:
```
{
	"Result": {
		"ResponseCode": "1076",
		"ResponseMessage": "Provider unable to process request",
		"ResponseTimestamp": "04112017091041",
		"MerchantID": "ANTERIOR001",
		"MerchantTxnID": "1104L703",
		"PaymentID": "107037767",
		"Currency": "USD",
		"Amount": "127000",
		"BillCurrency": null,
		"BillAmount": null,
		"CardHolder": "DMITRY FEDYUK",
		"CardType": "MASTERCARD",
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"ProviderResps": [
		{
			"ProviderName": "westpac",
			"ProviderOrder": "1",
			"ProviderResponse": "QA",
			"ProviderReference": null,
			"ResponseCode": "1076",
			"ResponseTimestamp": null
		}
	],
	"Country": {
		"CountryName": null,
		"CountryCode": null
	},
	"MethodResult": {
		"AuthCode": null,
		"CVVResult": null,
		"AVSStreetResult": null,
		"AVSZipResult": null,
		"SecureToken": null,
		"Status": "3",
		"AmountReceived": null,
		"RedirectUrl": null,
		"FeeAmount": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null,
		"ACSUrl": null,
		"PAReq": null
	},
	"ThreeDSecure": {
		"AuthenticationValue": null,
		"AuthenticationStatus": null,
		"Enrolled": null,
		"XID": null,
		"ECI": null,
		"PARes": null,
		"CAVVAlgorithm": null,
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"TokenResult": {
		"Custom1": null,
		"Custom2": null,
		"Custom3": null,
		"Custom4": null,
		"Custom5": null
	},
	"PaymentInfo": {
		"PaymentBrand": "MASTERCARD",
		"PaymentBrandType": "CREDIT",
		"PaymentBrandCategory": "CORPORATE",
		"PaymentIssuer": null,
		"PaymentIssuerCountry": "AU",
		"BIN": "531357"
	},
	"FraudResult": {
		"RiskCategory": null,
		"Score": null,
		"Status": null
	},
	"FraudChecks": [],
	"GeoChecks": [],
	"LoyaltyResult": null
}
```

### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/auth
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Sat, 04 Nov 2017 09:10:39 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: application/json; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Content-Length: 1498
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8,ru;q=0.6
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 2978
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=55AACA4E5E96284CF1126EAC44D93FED6FC47A665B1320F31ECC49FBFE4C28480F64CDD5908A1808A050A2DA2D9974B0E592DE9DF15BFC19FAD5718DCE264C6AE024B7046B7C780DEE1A596F44FD0CB532C866AEF9ADF67A36C5D0D0F2191615D56D282022ECF263F33919ACDCE70BAA3D12056DF19B5A9F30D0FF8F25EFA76EFD8B08364FF7D795133DEE8C315CB5D12C46F856F7616398D7F9098800B71BC1B83BE665E163AA8C923B4904974DC457246D8683; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://hubuat.alphacommercehub.com.au
Pragma: no-cache
Referer: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36
X-APC-MerchantTxnID: 1104L703
X-APC-PPSessionId: 358617d7-7b4f-4b48-bd87-3facae901db8
X-Requested-With: XMLHttpRequest
```
### Form data
```
cardHolder: DMITRY FEDYUK
cardNumber: 5313 5700 0000 0011
cardExpiryDate: 07 / 2020
cardCVV: 123
country: AU
street1: 72 Liverpool St
street2:
city: Sydney
zip: 2000
emailAddress:
token: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJwYXlsb2FkIjoie1wiVXNlcklkXCI6XCIxMjNcIixcIk1lcmNoYW50SURcIjpcIkFOVEVSSU9SMDAxXCIsXCJNZXJjaGFudFR4bklEXCI6XCIxMTA0TDcwM1wiLFwiQ3VycmVuY3lcIjpcIlVTRFwiLFwiQW1vdW50XCI6XCIxMjcwMDBcIixcIlRyYW5zYWN0aW9uVGltZXN0YW1wXCI6bnVsbCxcIk1ldGhvZFwiOlwiQ0NcIixcIkxhbmd1YWdlXCI6XCJlblwiLFwiVHhuRGV0YWlsc1wiOm51bGwsXCJDYXB0dXJlXCI6XCJZXCIsXCJFbWFpbEFkZHJlc3NcIjpcImRmZWRpdWtAZ21haWwuY29tXCIsXCJJUEFkZHJlc3NcIjpcIjkyLjI0My4xNjYuOFwiLFwiVGVsTm9cIjpcIjYxMjgyNjg4ODg4XCIsXCJDdXN0b21lcklEXCI6XCJkZmVkaXVrQGdtYWlsLmNvbVwiLFwiU29jaWFsSURcIjpcIjc0MDQxODAzNjgzXCIsXCJTdHJlZXQxXCI6XCI3MiBMaXZlcnBvb2wgU3RcIixcIlN0cmVldDJcIjpudWxsLFwiQ2l0eVwiOlwiU3lkbmV5XCIsXCJaaXBcIjpcIjIwMDBcIixcIlN0YXRlXCI6XCJNYWdlMi5QUk9cIixcIkNvdW50cnlcIjpcIkFVXCIsXCJNZXJjaGFudERlc2NyaXB0b3JcIjpcIk1hZ2UyLlBST1wiLFwiU3VjY2Vzc1VSTFwiOlwiaHR0cHM6Ly9tYWdlMi5wcm8vc2FuZGJveC9kZmUtYWxwaGFjb21tZXJjZWh1Yi9jdXN0b21lclJldHVyblwiLFwiVGhyZWVEU2VjdXJlQnlwYXNzXCI6XCJZXCIsXCJDYXJkVHlwZVwiOlwiYW1leCxqY2IsbWFzdGVyY2FyZCxkaXNjb3ZlcixtYXN0ZXJjYXJkZGViaXQsdmlzYSx2aXNhZGViaXQsdW5pb25wYXksbWFlc3Rybyx1YXRwLGRpbmVyc2NsdWJcIixcIkZyYXVkQ2hlY2tCeXBhc3NcIjpcIk5cIixcIkluc3RhbGxtZW50XCI6bnVsbCxcIlBheUxhdGVyTW9udGhzXCI6bnVsbCxcIlBheU51bWJlck1vbnRoc1wiOm51bGwsXCJJbnN0YWxsbWVudElEXCI6bnVsbCxcIkluc3RhbGxtZW50VHlwZVwiOm51bGwsXCJDYW5jZWxVUkxcIjpcImh0dHBzOi8vbWFnZTIucHJvL3NhbmRib3gvZGZlLWFscGhhY29tbWVyY2VodWIvY3VzdG9tZXJSZXR1cm5cIixcIkNhcmROdW1iZXJcIjpudWxsLFwiQ2FyZEhvbGRlclwiOm51bGwsXCJDYXJkRXhwaXJ5TW9udGhcIjpudWxsLFwiQ2FyZEV4cGlyeVllYXJcIjpudWxsLFwiQ2FyZENWVlwiOm51bGwsXCJDaGFubmVsVHlwZVwiOlwiMDdcIixcIlNlY3VyZVRva2VuXCI6bnVsbCxcIlNlc3Npb25JZFwiOm51bGwsXCJUYXhBZGRlZFwiOmZhbHNlLFwiVGF4UmF0ZVwiOjAuMCxcIk5vdGlmaWNhdGlvbkVtYWlsXCI6bnVsbCxcIlJldHVyblRleHRcIjpudWxsLFwiUmVwbHlVcmxcIjpudWxsLFwiVXBkYXRlVVJMXCI6bnVsbCxcIk9yaWdpblwiOlwiSFBQXCIsXCJUcmFuc2FjdGlvblR5cGVcIjpudWxsLFwiQ3VzdG9tMVwiOm51bGwsXCJQYXlXaXRoUG9pbnRzXCI6bnVsbCxcIk1lcmNoYW50RGF0YVwiOm51bGwsXCJUZXJtaW5hbFwiOm51bGwsXCJQYXltZW50UHJvY2Vzc29yQVBJSWRcIjpudWxsLFwiT3JkZXJEZXRhaWxzXCI6W3tcIkl0ZW1OYW1lXCI6bnVsbCxcIkl0ZW1RdWFudGl0eVwiOjAsXCJJdGVtQW1vdW50XCI6bnVsbCxcIkl0ZW1QcmljZVwiOm51bGwsXCJJdGVtVGF4RXhlbXB0XCI6ZmFsc2V9LHtcIkl0ZW1OYW1lXCI6bnVsbCxcIkl0ZW1RdWFudGl0eVwiOjAsXCJJdGVtQW1vdW50XCI6bnVsbCxcIkl0ZW1QcmljZVwiOm51bGwsXCJJdGVtVGF4RXhlbXB0XCI6ZmFsc2V9LHtcIkl0ZW1OYW1lXCI6bnVsbCxcIkl0ZW1RdWFudGl0eVwiOjAsXCJJdGVtQW1vdW50XCI6bnVsbCxcIkl0ZW1QcmljZVwiOm51bGwsXCJJdGVtVGF4RXhlbXB0XCI6ZmFsc2V9XSxcIkN1c3RvbUZpZWxkc1wiOltdLFwiQWNjZXB0M0RTQXV0aGVudGljYXRpb25VbmF2YWlsYWJsZVwiOmZhbHNlLFwiQWNjZXB0M0RTRW5yb2xCeXBhc3NcIjpmYWxzZSxcIlBQU2Vzc2lvbklkXCI6XCI3ZDQ4MGNlZi0zYjBmLTQ1YzktOTk1Yy1mMjZkNTI5YzVjNjVcIn0ifQ.PhfOLgnJMEXRXJR24y4dc4YnAQOSCjOTl9LyerMkm6M
cardExpiryMonth: 7
cardExpiryYear: 2020
payPageID: ac580978-81f0-4716-bfc3-3bc46eab8603
merchantSequenceNumber:
clientIP: 92.243.166.8
```

**Comment by dmitrii-fediuk** [comment]
## The initial request
### General
```
Request URL: https://hubuat.alphacommercehub.com.au/pp/ac580978-81f0-4716-bfc3-3bc46eab8603
Request Method: POST
Status Code: 200 OK
Remote Address: 76.12.61.237:443
Referrer Policy: no-referrer-when-downgrade
```

### Response headers
```
HTTP/1.1 200 OK
Date: Sat, 04 Nov 2017 09:09:54 GMT
Server: Microsoft-IIS/8.5
Cache-Control: private
Content-Type: text/html; charset=utf-8
X-AspNetMvc-Version: 4.0
X-AspNet-Version: 4.0.30319
X-Powered-By: ASP.NET
Vary: Accept-Encoding
Content-Encoding: gzip
Content-Length: 4924
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
```

### Request headers
```
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.8,ru;q=0.6
Cache-Control: no-cache
Connection: keep-alive
Content-Length: 1080
Content-Type: application/x-www-form-urlencoded
Cookie: __RequestVerificationToken=n-B5epNpZExPBpdLRWY_IOg_zxlnxkKjL1utioq1JrxE6rgCQN9BJ480lFpp8mWnGdzLeRMhjjnUk-PHF_BJBywBGWw7qNJYr6zSdMzZPcjvYobCiCRY5Eyc2vjI9cd0Pdv0_Q2; _timeZoneOffset=180; _ga=GA1.3.685466192.1508711293; ASP.NET_SessionId=gam2fam11wnk5cryudsqbxps; .ASPXAUTH=55AACA4E5E96284CF1126EAC44D93FED6FC47A665B1320F31ECC49FBFE4C28480F64CDD5908A1808A050A2DA2D9974B0E592DE9DF15BFC19FAD5718DCE264C6AE024B7046B7C780DEE1A596F44FD0CB532C866AEF9ADF67A36C5D0D0F2191615D56D282022ECF263F33919ACDCE70BAA3D12056DF19B5A9F30D0FF8F25EFA76EFD8B08364FF7D795133DEE8C315CB5D12C46F856F7616398D7F9098800B71BC1B83BE665E163AA8C923B4904974DC457246D8683; email=seb%40anterior.com.au
Host: hubuat.alphacommercehub.com.au
Origin: https://localhost.com:900
Pragma: no-cache
Referer: https://localhost.com:900/store/checkout/
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36
```

### Form data
```
3DSecureBypass: Y
Amount: 127000
CancelURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
Capture: Y
City: Sydney
Country: AU
Currency: USD
CustomerID: dfediuk@gmail.com
EmailAddress: dfediuk@gmail.com
FraudCheckBypass: N
IPAddress: 92.243.166.8
Language: en
MerchantDescriptor: Mage2.PRO
MerchantID: ANTERIOR001
MerchantTxnID: 1104L703
Method: CC
OrderDetails[0][ItemAmount]: 36000
OrderDetails[0][ItemName]: Alligator Briefcase
OrderDetails[0][ItemQuantity]: 2
OrderDetails[0][ItemTaxExempt]: TRUE
OrderDetails[1][ItemAmount]: 45000
OrderDetails[1][ItemName]: New Very Prive-41-Black
OrderDetails[1][ItemQuantity]: 1
OrderDetails[1][ItemTaxExempt]: TRUE
OrderDetails[2][ItemAmount]: 10000
OrderDetails[2][ItemName]: Shipping
OrderDetails[2][ItemQuantity]: 1
OrderDetails[2][ItemTaxExempt]: TRUE
SocialID: 74041803683
State: Mage2.PRO
Street1: 72 Liverpool St
SuccessURL: https://mage2.pro/sandbox/dfe-alphacommercehub/customerReturn
TelNo: 61282688888
UserId: 123
Zip: 2000
```

**Comment by dmitrii-fediuk** [comment]
Today I have got a response from the AlphaCommerceHub team:

> The test merchant is going to Westpac test host which only supports `AUD` and `NZD`.
`USD` was submitted. Please **change to `AUD`** and try again

**Comment by dmitrii-fediuk** [comment]
See also my generic unanswered question: [[AlphaCommerceHub] Which currencies are allowed by each payment option?](https://mage2.pro/t/4820)

**Comment by dmitrii-fediuk** [comment]
The `AUD ` payment currency does not help:

[AlphaHPP without 3D Secure (`3DSecureBypass` = `Y`) and with `Currency` = `AUD`: «Provider unable to process request» / «"ProviderName": "westpac"»](https://github.com/mage2pro/alphacommercehub/issues/12)

**Comment by dmitrii-fediuk** [comment]
It is fixed by #17 (The last 3 digits of every payment amount should be «000» in the test mode for Westpac).
